common/server_protocolhandler.cpp

```diff
@@ -765,3 +751,24 @@ void Server_ProtocolHandler::resetIdleTimer()
     lastActionReceived = timeRunning;
     idleClientWarningSent = false;
 }
+
+bool Server_ProtocolHandler::isUserPermittedToJoinRoom(const QString roomPermission, const QString roomPrivilegeLevel)
```

Comment:
There is no such thing as an admin/moderators.  admin and moderators are allowed to enter all rooms regardless of their privilege level.